Sources/Testing/Issues/Issue.swift

```diff
@@ -179,3 +179,217 @@ extension Issue.Kind: CustomStringConvertible {
+    /// property is the empty array.
+    public var comments: [Comment]
+
+    /// A ``SourceContext`` indicating where and how this issue occurred.
+    public var sourceContext: SourceContext
+
+    /// Whether or not this issue is known to occur.
+    public var isKnown = false
+
+    /// Initialize an issue instance with the specified details.
+    ///
+    ///
+    init(issue: Issue) {
+      self.kind = Issue.Kind.Snapshot(kind: issue.kind)
+      self.comments = issue.comments
+      self.sourceContext = issue.sourceContext
+      self.isKnown = issue.isKnown
+    }
+  }
+}
+
+extension Issue.Kind {
+  /// Serializable kinds of issues which may be recorded.
+  @_spi(ExperimentalSnapshotting)
+  public enum Snapshot: Sendable, Codable {
+    /// An issue which occurred unconditionally, for example by using
+    /// ``Issue/record(_:fileID:filePath:line:column:)``.
+    case unconditional
+
+    /// An issue due to a failed expectation, such as those produced by
+    /// ``expect(_:_:sourceLocation:)``.
+    ///
+    /// - Parameters:
+    ///   - expectation: The expectation that failed.
+    case expectationFailed(_ expectation: Expectation.Snapshot)
+
+    /// An issue due to a confirmation being confirmed the wrong number of
+    /// times.
+    ///
+    /// - Parameters:
+    ///   - actual: The number of times ``Confirmation/confirm(count:)`` was
+    ///     actually called.
+    ///   - expected: The expected number of times
+    ///     ``Confirmation/confirm(count:)`` should have been called.
+    ///
+    /// This issue can occur when calling
+    /// ``confirmation(_:expectedCount:fileID:filePath:line:column:_:)`` when
+    /// the confirmation passed to these functions' `body` closures is confirmed
+    /// too few or too many times.
+    case confirmationMiscounted(actual: Int, expected: Int)
+
+    /// An issue due to an `Error` being thrown by a test function and caught by
+    /// the testing library.
+    ///
+    /// - Parameters:
+    ///   - error: The error which was associated with this issue.
+    case errorCaught(_ errorDescription: String)
+
+    /// An issue due to a test reaching its time limit and timing out.
+    ///
+    /// - Parameters:
+    ///   - timeLimitComponents: The time limit reached by the test.
+    ///
+    /// @Comment {
+    ///   - Bug: The associated value of this enumeration case should be an
+    ///     instance of `Duration`, but the testing library's deployment target
+    ///     predates the introduction of that type.
+    /// }
+    case timeLimitExceeded(timeLimitComponents: (seconds: Int64, attoseconds: Int64))
+
+    /// A known issue was expected, but was not recorded.
+    case knownIssueNotRecorded
+
+    /// An issue occurred due to misuse of the testing library.
+    case apiMisused
+
+    /// An issue due to a failure in the underlying system, not due to a failure
+    /// within the tests being run.
+    case system
+
+    init(kind: Issue.Kind) {
```

Comment:
Simple casting initializers like this one should have unlabelled arguments, so: `init(_ kind: Issue.Kind)`